packages/mui-private-theming/src/ThemeProvider/ThemeProvider.js

```diff
@@ -52,14 +52,19 @@ function ThemeProvider(props) {
   }
 
   const theme = React.useMemo(() => {
-    const output = outerTheme === null ? localTheme : mergeOuterLocalTheme(outerTheme, localTheme);
+    let internalTheme = localTheme;
```

Comment:
Should we move this logic inside the `ThemeProvider` in @mui/system? This package was created initially only to serve as a common context for @mui/system & @mui/styles, it shouldn't contain anything that is specific to only one of the packages.